overcast.py

```diff
@@ -7,51 +7,105 @@
         r = self.session.post('https://overcast.fm/login', {'email': email, 'password': password})
         doc = lxml.html.fromstring(r.content)
         alert = doc.cssselect('div.alert')
         if alert:
             raise Exception("Can't login: {}".format(alert[0].text_content().strip()))
-            
+
     def _get_html(self, url):
         return lxml.html.fromstring(self.session.get(url).content)
-            
+
     def get_active_episodes(self):
         doc = self._get_html('https://overcast.fm/podcasts')
         return [
             self.get_episode_detail(cell.attrib['href'])
             for cell in doc.cssselect('a.episodecell')
             if 'href' in cell.attrib
         ]
-        
+
     def get_episode_detail(self, episode_id):
         episode_href = urlparse.urljoin('https://overcast.fm', episode_id)
         doc = self._get_html(episode_href)
+
+        time_elapsed_seconds = int(doc.cssselect('audio#audioplayer')[0].attrib['data-start-time'])
+        time_remaining_seconds = self.get_episode_time_remaining_seconds(episode_id, doc)
+        duration = time_elapsed_seconds + time_remaining_seconds
+        if time_elapsed_seconds == duration:
+            duration = 0
+
         return {
             'id': episode_href.lstrip('/'),
             'title': doc.cssselect('div.titlestack div.title')[0].text_content(),
             'podcast_title': doc.cssselect('div.titlestack div.caption2 a')[0].text_content(),
-            'duration':  60, # fixme - fuck where do I get the duration from :(
+            'offsetMillis': time_elapsed_seconds * 1000,
+            'duration': duration,
+            'data_item_id': doc.cssselect('audio#audioplayer')[0].attrib['data-item-id'],
+            'data_sync_version': doc.cssselect('audio#audioplayer')[0].attrib['data-sync-version'],
+            'albumArtURI': doc.cssselect('div.fullart_container img')[0].attrib['src'],
             'audio_uri': doc.cssselect('audio#audioplayer source')[0].attrib['src'],
             'audio_type': doc.cssselect('audio#audioplayer source')[0].attrib['type'],
         }
-    
+
+    def get_episode_time_remaining_seconds(self, episode_id, episode_html):
+        log.debug('''getting the remaining time. episode id is %s''', episode_id)
+        podcast_id = episode_html.cssselect('div.titlestack div.caption2 a')[0].attrib['href']
+        podcast_href = urlparse.urljoin('https://overcast.fm', podcast_id)
+        doc = self._get_html(podcast_href)
+        time_remaining_seconds = 0
+
+        for cell in doc.cssselect('a.extendedepisodecell'):
+            if episode_id in cell.attrib['href']:
+                cell.cssselect('div.caption2')[0].text_content()
+                unparsed_time_remaining = cell.cssselect('div.singleline')[1].text_content()
+                time_remaining_seconds = self.duration_in_seconds(unparsed_time_remaining)
+                break
+
+        return time_remaining_seconds
+
+    def duration_in_seconds(self, str):
```

Comment:
This could probably be refactored to be a standalone function rather than a method - it's not really using `self`, so probably could stand alone. 

I'm also not entirely sure I understand the logic here, so just a note to myself that I should figure out exactly what's up so I can debug this better in the future.